hw/ip/pwrmgr/data/pwrmgr_testplan.hjson

```diff
@@ -146,14 +146,28 @@
               until it becomes 1.
             - Checks the `reset_status` CSRs.
             - Checks `ip_clk_en` output has a low transition.
-            - Checks the output `pwr_rst_req.reset_cause` matches a reset
-              cause.
+            - Checks the output `pwr_rst_req.reset_cause` matches HwReq.
             - Checks the output `pwr_rst_req.rstreqs` matches the enabled
               resets.
             '''
       milestone: V2
       tests: []
     }
+    {
+      name: main_power_glitch
+      desc: '''
+            Test that a main power glitch triggers a reset request.
+
+            **Stimulus**:
+            - Drop the AST input main_pok.
```

Comment:
Do we need to cross this with various PWR FSM states? For example, what happens if `pok` drops while pwrmgr is progressing through low power entry or exit. 

@tjaychen

Reply:
The logic depends on the state in some cases. Perhaps we should FPV this? Another way to test it is to flip pok while transitioning in and out of low power with main_pd_n on and off...

Reply:
in theory, the same behavior applies as with all other reset requests.
if you are already in the middle of handling something, then the reset request is handled on the next cycle. The pok reset request is not handled any differently compared to other reset requests with the exception of always being enabled.

Reply:
I suppose then our strategy should be to use `{rstreqs, pok}` wide input vector for the reset req tests?

Reply:
Good thinking. And some can be disabled, while others (like this) are always on.